### ISSUE TITLE
DL3009 forces `rm -fr /var/lib/apt/lists/*` and doesn't accept `rm -fr /var/lib/apt/lists/*`

### ISSUE THREAD

**Issue by paul-a-kennedy-rally** [hadolint/hadolint#872]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
I expect the command to remove directory to be sufficient without forcing the shell expansion of its contents.

### Actual behavior
```
Dockerfile:28 DL3009 info: Delete the apt-get lists after installing something
```

### Steps to reproduce the behavior

```
RUN apt-get update && \
    apt-get install -y ca-certificates curl --no-install-recommends && \
    apt-get clean && \
    rm -fr /var/lib/apt/lists
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
$ hadolint --version 
Haskell Dockerfile Linter UNKNOWN
```

**Dockerfile** (if relevant)

```Dockerfile
(paste content of your Dockerfile)
```

**Additional environment details (OS, stack version, etc.)**

It is possible to have a command line too long with a shell expansion like the one this rule is requiring (ask me how I know).

**Comment by Hongbo-Miao** [comment]
Just met same issue. My sample **Dockerfile** looks like

```dockerfile
FROM docker.io/eqalpha/keydb:x86_64_v6.2.2

RUN apt-get update \
  && apt-get upgrade -y \
  && apt-get install libgomp1 -y --no-install-recommends \
  && apt-get clean \
  && rm -r -f /var/lib/apt/lists/

RUN ls
```

haolint will give me error

> DL3009 info: Delete the apt-get lists after installing something

for `rm -r -f /var/lib/apt/lists/`

However, if I changed back to `rm -r -f /var/lib/apt/lists/*`, it won't.

But the difference between is that `rm -r -f /var/lib/apt/lists/` removes the /var/lib/apt/lists/ directory itself, along with all of its contents, while `rm -r -f /var/lib/apt/lists/*` removes all the files and directories in the /var/lib/apt/lists/ directory.

I did more research, if I change my **Dockerfile** to

```dockerfile
FROM docker.io/eqalpha/keydb:x86_64_v6.2.2
RUN ls
```

and then sh inside by

```shell
docker build --tag=my-tag .
docker run --rm -it --entrypoint sh my-tag
```

By default, `/var/lib/apt/lists/` is there and empty.

So I am thinking one "benefit" using `rm -r -f /var/lib/apt/lists/*` will recover to original status without removing the directory itself.

On the other side, I think it is not a big deal though even removing this folder itself. As even this folder got deleted, after `apt-get update && apt-get upgrade -y`, it will still come back. Unless some scripts expecting this folder even empty to be there which might causes issues.
